simulation/simple_sensor_simulator/test/src/test_scenario_simulator.cpp

```diff
@@ -0,0 +1,340 @@
+#include <simulation_interface/conversions.hpp>
+#include <simulation_interface/zmq_multi_client.hpp>
+#include <traffic_simulator/helper/helper.hpp>
+
+#include "sensor_simulation/expect_eq_macros.hpp"
+
+auto makeInitializeRequest() -> simulation_api_schema::InitializeRequest
+{
+  auto request = simulation_api_schema::InitializeRequest();
+  request.set_lanelet2_map_path(
+    ament_index_cpp::get_package_share_directory("traffic_simulator") + "/map/lanelet2_map.osm");
+  return request;
+}
+
+auto makeUpdateFrameRequest() -> simulation_api_schema::UpdateFrameRequest
+{
+  auto request = simulation_api_schema::UpdateFrameRequest();
+  request.set_current_scenario_time(1.0);
+  request.set_current_simulation_time(1.0);
+  return request;
+}
+
+auto makeDespawnEntityRequest(const std::string name = "name")
+  -> simulation_api_schema::DespawnEntityRequest
+{
+  auto request = simulation_api_schema::DespawnEntityRequest();
+  request.set_name(name);
+  return request;
+}
+
+auto makeSpawnVehicleEntityRequest(const std::string name = "name", const bool is_ego = false)
+  -> simulation_api_schema::SpawnVehicleEntityRequest
+{
+  auto request = simulation_api_schema::SpawnVehicleEntityRequest();
+  const auto params = traffic_simulator_msgs::msg::VehicleParameters{};
+  simulation_interface::toProto(params, *request.mutable_parameters());
+  request.mutable_parameters()->set_name(name);
+  request.set_is_ego(is_ego);
+  return request;
+}
+
+auto makeSpawnPedestrianEntityRequest(const std::string name = "name")
+  -> simulation_api_schema::SpawnPedestrianEntityRequest
+{
+  auto request = simulation_api_schema::SpawnPedestrianEntityRequest();
+  const auto params = traffic_simulator_msgs::msg::PedestrianParameters{};
+  simulation_interface::toProto(params, *request.mutable_parameters());
+  request.mutable_parameters()->set_name(name);
+  return request;
+}
+
+auto makeSpawnMiscObjectEntityRequest(const std::string name = "name")
+  -> simulation_api_schema::SpawnMiscObjectEntityRequest
+{
+  auto request = simulation_api_schema::SpawnMiscObjectEntityRequest();
+  const auto params = traffic_simulator_msgs::msg::MiscObjectParameters{};
+  simulation_interface::toProto(params, *request.mutable_parameters());
+  request.mutable_parameters()->set_name(name);
+  return request;
+}
+
+auto makeAttachDetectionSensorRequest() -> simulation_api_schema::AttachDetectionSensorRequest
+{
+  auto request = simulation_api_schema::AttachDetectionSensorRequest();
+  auto configuration = traffic_simulator::helper::constructDetectionSensorConfiguration(
+    "entity_name", "awf/universe", 1.0);
+  *request.mutable_configuration() = configuration;
+  return request;
+}
+
+auto makeAttachLidarSensorRequest() -> simulation_api_schema::AttachLidarSensorRequest
+{
+  auto request = simulation_api_schema::AttachLidarSensorRequest();
+  auto configuration = traffic_simulator::helper::constructLidarConfiguration(
```

Comment:
It looks like the functions makeAttachDetectionSensorRequest and makeAttachLidarSensorRequest are using helper functions from the traffic_simulator package to construct configurations. This dependency could cause tests in the scenario_simulator to fail if there are changes in the traffic_simulator package, rather than changes in the simulation_api_schema.pb.h.

Given that only two helper functions are involved, it seems reasonable to duplicate these functions within the simple_sensor_simulator package. This would ensure that the tests fail solely due to changes in the simulation_api_schema.pb.h without being affected by modifications in the traffic_simulator package.

Reply:
We can solve this similarly to how we handle makeAttachOccupancyGridSensorRequest by defining the configuration directly within the simple_sensor_simulator package.

Reply:
I propose to add new directory where we can put a file with definition of constructLidarConfiguration and other useful functions. Take a look here [PR](https://github.com/RobotecAI/scenario_simulator_v2/pull/26)